Task2/MySimpleLazy.cs

```diff
@@ -0,0 +1,21 @@
+using System;
+class MySimpleLazy<T> : ILazy<T>
+{
+    private Func<T> supplier;
+    private T value;
+    private bool isCreatedValue = false;
+
+    public MySimpleLazy(Func<T> supplier){
+        this.supplier = supplier;
+    }
+
+    public T Get()
+    {
+       if (!isCreatedValue){
+        value = supplier();
+        supplier = null;
+        isCreatedValue = true;
+       }
```

Comment:
```suggestion
       if (!isCreatedValue){
           value = supplier();
           supplier = null;
           isCreatedValue = true;
       }
```
не забываем про форматирование